### PR TITLE
cargo-edit: 0.2.0 -> 0.3.0

### PR DESCRIPTION
**Pull request by alyssais**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Mic92** [comment]
oh, 3000 lines nix code for a single packages?

**Comment by Mic92** [comment]
Does it also compile with `buildRustPackage`?

**Comment by alyssais** [comment]
> Does it also compile with `buildRustPackage`?

Seems to!

**Comment by Mic92** [comment]
@GrahamcOfBorg build cargo-edit

File: pkgs/tools/package-management/cargo-edit/default.nix [hunk 1]
```diff
@@ -1,27 +1,25 @@
...
+    owner = "killercup";
+    repo = "cargo-edit";
+    rev = "v${version}";
+    sha256 = "0ngxyzqy5pfc0fqbvqw7kd40jhqzp67qvpzvh3yggk9yxa1jzsp0";
+  };
+
+  cargoSha256 = "1j7fqswdx6f2i5wr0pdavdvcv18j1l27a8ighr75p7f54apa27l8";
 
-      propagatedBuildInputs = stdenv.lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.Security ];
+  propagatedBuildInputs = lib.optional stdenv.isDarwin darwin.apple_sdk.frameworks.Security;
```
Comment (Mic92) [inline pkgs/tools/package-management/cargo-edit/default.nix]: On linux `cargo-edit` also needs `pkgconfig` in `nativeBuildInputs` and `openssl` in `buildInputs`.

Comment (Mic92) [reply pkgs/tools/package-management/cargo-edit/default.nix]: Any reason why `darwin.apple_sdk.frameworks.Security` is not in `buildInputs`?

Comment (Mic92) [reply pkgs/tools/package-management/cargo-edit/default.nix]: `darwin.apple_sdk.frameworks.Security` is probably fine. `pkgs/build-support/rust/default-crate-overrides.nix` does the same.

**Comment by Mic92** [comment]
@GrahamcOfBorg build cargo-edit

**Comment by Mic92** [comment]
Thanks!

**Pull request merged**